software/controller/lib/core/system_constants.h

```diff
@@ -0,0 +1,92 @@
+
+// Blower is hooked up to PB3 (see [PCB]),
+// which can be linked to Timer2 chanel 2 using AF1 (see [DS], p77)
+static constexpr GPIO::PwmChannel BlowerChannel = {GPIO::Port::B, 3, GPIO::AlternativeFunction::AF1,
+                                                   PeripheralID::Timer2, 2};
+// Blower is driven by a 20kHz PWM, as a compromise between resolution and response time
+static constexpr Frequency BlowerFreq = kilohertz(20);
+
+// psol is hooked up to PA11 (see [PCB]),
+// which can be linked to Timer1 chanel 4 using AF1 (see [DS], p76)
+static constexpr GPIO::PwmChannel PSolChannel = {GPIO::Port::A, 11, GPIO::AlternativeFunction::AF1,
+                                                 PeripheralID::Timer1, 4};
+// Psol is driven by a 5kHz PWM (\TODO: find rationale behind this?)
+static constexpr Frequency PSolFreq = kilohertz(5);
+// Testing in Edwin's garage, we found that the psol was fully closed at
+// somewhere between 0.75 and 0.80 (i.e. definitely zero at 0.75 and probably
+// zero a bit above that) and fully open at 0.90.
+// \TODO: the values in the comment are inconsistent with the code, have Edwin confirm those.
+static constexpr float PSolClosed = 0.35f;
+static constexpr float PSolOpen = 0.75f;
+
+/**************************************************************************************
+ *                                 PHYSICAL CONSTANTS                                 *
+ **************************************************************************************/
+
+// Diameters and correction coefficient relating to 3/4in Venturi, see https://bit.ly/2ARuReg.
+// Correction factor of 0.97 is based on ISO recommendations for Reynolds of roughly 10^4 and
+// machined (rather than cast) surfaces. Data fit is in good agreement based on comparison to
+// Fleisch pneumotachograph; see https://github.com/RespiraWorks/Ventilator/pull/476
+static constexpr Length VenturiPortDiameter{millimeters(15.05f)};
+static constexpr Length VenturiChokeDiameter{millimeters(5.5f)};
+static constexpr float VenturiCorrection{0.97f};
+
+static_assert(VenturiPortDiameter > VenturiChokeDiameter);
+static_assert(VenturiChokeDiameter > meters(0));
+
+/**************************************************************************************
+ *                                FUNCTIONAL CONSTANTS                                *
+ **************************************************************************************/
+
+// Loop period of our main controller loop
+static constexpr Duration MainLoopPeriod = milliseconds(10);
+
+// Transition from PEEP to PIP pressure hapens over this length of time.  Citation:
+// https://respiraworks.slack.com/archives/C011CJQV4Q7/p1591763842312500?thread_ts=1591759016.310200&cid=C011CJQV4Q7
+inline constexpr Duration RiseTime = milliseconds(100);
+
+// TODO: VolumeIntegrationInterval (for volume integrator) was not chosen carefully.
+static constexpr Duration VolumeIntegrationInterval = milliseconds(5);
```

Comment:
And another section could be something like "pneumatic routing". Though, again, we can defer doing this for later.